hle-survey/transformation/src/main/scala/org/broadinstitute/monster/dap/HealthTransformations.scala

```diff
@@ -64,14 +65,109 @@ object HealthTransformations {
             categorical
           )
       }
-    } else {
-      None
-    }
+    } else None
 
-  // list conditions and assign categorical Longs
-  val infectiousDiseaseCondition = 1L
+  def mapCongenitalEyeDisorder(rawRecord: RawRecord): Iterable[HlesHealthCondition] =
+    if (rawRecord.getBoolean("hs_congenital_yn") && rawRecord.getBoolean("hs_cg_eye_disorders_yn"))
+      eye_cg_disorders.flatMap {
+        // "other" case
+        case ("other", categorical) =>
+          createHealthConditionRow(
+            rawRecord,
+            "cg_eye_other",
+            eyeDiseaseCondition,
+            categorical,
+            isCongenital = true,
+            conditionOtherDescription = rawRecord.getOptional("hs_cg_eye_other_spec")
+          )
+        // general case
+        case (disorder, categorical) =>
+          createHealthConditionRow(
+            rawRecord,
+            s"cg_eye_${disorder}",
+            eyeDiseaseCondition,
+            categorical,
+            isCongenital = true
+          )
+      }
+    else None
+
+  def mapEyeDisease(rawRecord: RawRecord): Iterable[HlesHealthCondition] =
+    if (rawRecord.getBoolean("hs_dx_eye_yn")) {
+      eye_diseases.flatMap {
+        // "other" case
+        case ("eye_other", categorical) =>
+          createHealthConditionRow(
+            rawRecord,
+            "dx_eye_other",
+            eyeDiseaseCondition,
+            categorical,
+            conditionOtherDescription = rawRecord.getOptional("hs_dx_eye_other_spec")
+          )
+        case ("blind", categorical) => {
+          val isCauseKnown = rawRecord.getBoolean("hs_dx_eye_cause_yn")
+          val conditionCause =
+            if (isCauseKnown) rawRecord.getOptionalNumber("hs_dx_eye_cause") else None
+          createHealthConditionRow(
+            rawRecord,
+            "dx_blind",
+            eyeDiseaseCondition,
+            categorical,
+            conditionCause = conditionCause,
+            conditionCauseOtherDescription =
+              if (isCauseKnown && (conditionCause.getOrElse(None) == 98))
+                rawRecord.getOptional("hs_dx_eye_cause_other")
+              else None
+          )
+        }
+        case (disease, categorical) =>
+          createHealthConditionRow(rawRecord, s"dx_${disease}", eyeDiseaseCondition, categorical)
+      }
+    } else None
+
+  ////////////////////////////////
+  //// condition categoricals ////
+  ////////////////////////////////
+  val eyeDiseaseCondition = 1L
+  val infectiousDiseaseCondition = 50L
+
+  //////////////////////////////////////
+  //// specific condition type maps ////
+  //////////////////////////////////////
+
+  // eye disorders and diseases
+  val eye_cg_disorders: Map[String, Long] = Map(
+    "blind" -> 0,
+    "cat" -> 1,
+    "glauc" -> 2,
+    "kcs" -> 3,
+    "ppm" -> 4,
+    "miss" -> 5,
+    "other" -> 98
```

Comment:
Only asking for my own understanding - Are these values based on the order of the Redcap questions? Or is this mapping defined somewhere that you're copying it over from?

Reply:
Based on order. We have to generate/define numerical categoricals for these conditions, and instead of like iterating over a list and using the index, we want to write them out so that we have some clearly defined human-readable source of truth for what the number for a given condition is.

Reply:
Great! Thanks, that makes sense.